internal/integritymonitor/integritymonitor.go

```diff
@@ -210,26 +208,18 @@ func integrityCheckFailed(
 	alertSender alerts.Sender,
 	kubeClient *services.KubeClient,
 ) {
-
-	var msg string
 	var path string
 	var integrityError *IntegrityError
 	if errors.As(err, &integrityError) {
 		path = integrityError.Path
-		switch integrityError.Type {
-		case ErrTypeFileMismatch:
-			msg = IntegrityMessageFileMismatch
-		case ErrTypeNewFile:
-			msg = IntegrityMessageNewFileFound
-		case ErrTypeFileDeleted:
-			msg = IntegrityMessageFileDeleted
-		}
-		log.WithError(err).WithField("path", integrityError.Path).Error("check integrity failed")
-	} else {
-		msg = err.Error()
-		log.WithError(err).Error("check integrity failed")
 	}
 
+	l := log.WithError(err)
+	if len(path) > 0 {
+		l = l.WithField("path", path)
```

Comment:
minor optimization: 
- you don't need `var path string` - l.WithField("path", integrityError.Path) - move it upper
- and probably `var integrityError *IntegrityError`  - ....As(err, &IntegrityError{})

Reply:
done